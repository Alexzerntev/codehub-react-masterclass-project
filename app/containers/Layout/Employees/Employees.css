.employees-container {
    display: flex;
    height: 80vh;
    margin: 2em 0;
    background: #fff;
    box-shadow: 0px 0px 10px 0px rgba(0, 0, 0, 0.75);
  }