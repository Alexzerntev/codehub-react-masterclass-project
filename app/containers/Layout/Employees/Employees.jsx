import React, { Component } from "react";
import { Row, Col } from "reactstrap";

import API from "../../../utils/API";
import EmployeeList from "./EmployeeList/EmployeeList";

import "./Employees.css";
import EmployeeDetails from "./EmployeeDetails/EmployeeDetails";

class Employees extends Component {
  state = {
    employees: [],
    selectedEmployee: null,
  };

  async componentDidMount() {
    const employees = await API.get("employees");

    this.setState({ employees: employees.data });
  }

  onSelectEmployee = (e, id) => {
    e.preventDefault();
    const selectedEmployee = this.state.employees.find((employee) => employee.id === id);

    this.setState({ selectedEmployee });
  };

  onInputChange = (e) => {
    const { value, name } = e.target;

    this.setState({ selectedEmployee: { ...this.state.selectedEmployee, [name]: value } });
  };

  onSelectChange = (e) => {
    const { value } = e.target;

    this.setState({ selectedEmployee: { ...this.state.selectedEmployee, gender: value } });
  };

  onFormSubmit = async (e) => {
    e.preventDefault();
    const { id, ...rest } = this.state.selectedEmployee;
    await API.put(`employees/${id}`, rest);
    const employees = await API.get("employees");

    this.setState({ employees: employees.data });
  };

  render() {
    const { employees, selectedEmployee } = this.state;

    return (
      <Row>
        <Col xs={12}>
          <div className="employees-container">
            <EmployeeList employees={employees} />
            <EmployeeDetails selectedEmployee={selectedEmployee} />
          </div>
        </Col>
      </Row>
    );
  }
}

export default Employees;
