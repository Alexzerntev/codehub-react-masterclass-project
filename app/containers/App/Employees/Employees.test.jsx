import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import Employees from "./Employees"

configure({ adapter: new Adapter() });

describe('Employees', () => {
    const employees = shallow(<Employees />);

    it('renders correctly', () => {
        expect(employees).toMatchSnapshot();
    });

    describe('Initialize', () => {
        it('initializes `employees` with an empty list of employees', () => {
            expect(employees.prop("employees")).toEqual([]);
        });
    });

});

