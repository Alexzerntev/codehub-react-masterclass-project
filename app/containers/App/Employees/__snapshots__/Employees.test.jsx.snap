// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employees renders correctly 1`] = `ShallowWrapper {}`;
